app/Models/Log.php

```diff
@@ -52,6 +52,51 @@ public function setLogDataAttribute($value)
         $this->attributes['log_data'] = serialize($value);
     }
 
+    public function dataForDisplay()
+    {
+        if ($this->log_operation === 'LOG_TOPIC_TYPE') {
+            switch ($this->log_data['type']) {
+                case 0:
+                    $translationString = 'unpin';
+                    break;
+                case 1:
+                    $translationString = 'pin';
+                    break;
+                case 2:
+                    $translationString = 'announcement';
+                    break;
+            }
+
+            return trans("forum.topic.logs.data.{$translationString}");
```

Comment:
forgot about this but `trans` shouldn't be used directly. Use `osu_trans` instead.